### PR TITLE
made the bee a basic insect

### PR DESCRIPTION
**Pull request by Ben10Omintrix**
## About The Pull Request
the bee now a baisc insect he will now go to  find his home and he will go and pollinated the plants and helped the queen make children by polliniting the plants and he will. the queen will leve the hive more rarely than the normal bees so she can stay in the hive to make kids

## Why It's Good For The Game
the bee now is a basic insect so it means he have a better ai

## Changelog
:cl:
balance: the bee now can fly over the machines so its easy for him to go to the hydroponics machine
fix: player bees now will not be stuck inside the hive if he entered it, they can now leave it
fix: fixed a har deleted when the hive is deleted all the bees still have a refence to the hive now its fixed
fix: now when a player interacted with the bee hive the bees will now leave the hive to defend the hive (it was glitched)
refactor: the bees now are a basic insect.
/:cl:

### CODE REVIEW

File: code/modules/research/experimentor.dm [hunk 1]
```diff
@@ -648,7 +648,7 @@
 		/mob/living/basic/pet/fox,
 		/mob/living/simple_animal/crab,
 		/mob/living/simple_animal/hostile/bear,
-		/mob/living/simple_animal/hostile/bee,
+		/mob/living/basic/bee,
```
Comment (san7890) [inline code/modules/research/experimentor.dm:651]: alphabetize this list.

File: code/modules/events/vent_clog.dm [hunk 2]
```diff
@@ -221,7 +221,7 @@
 /datum/round_event/vent_clog/major/get_mob()
 	var/static/list/mob_list = list(
 		/mob/living/basic/mouse/rat,
-		/mob/living/simple_animal/hostile/bee,
+		/mob/living/basic/bee,
```
Comment (san7890) [inline code/modules/events/vent_clog.dm:224]: alphabetize

File: code/modules/events/vent_clog.dm [hunk 3]
```diff
@@ -256,7 +256,7 @@
 /datum/round_event/vent_clog/critical/get_mob()
 	var/static/list/mob_list = list(
 		/mob/living/basic/carp,
-		/mob/living/simple_animal/hostile/bee/toxin,
+		/mob/living/basic/bee/toxin,
```
Comment (san7890) [inline code/modules/events/vent_clog.dm:259]: alphabetize

File: code/modules/hydroponics/beekeeping/beebox.dm [hunk 4]
```diff
@@ -255,9 +254,9 @@
 
 /obj/structure/beebox/deconstruct(disassembled = TRUE)
 	new /obj/item/stack/sheet/mineral/wood (loc, 20)
-	for(var/mob/living/simple_animal/hostile/bee/B in bees)
-		if(B.loc == src)
-			B.forceMove(drop_location())
+	for(var/mob/living/basic/worker in bees)
```
Comment (san7890) [inline code/modules/hydroponics/beekeeping/beebox.dm:257]: this spawns the abstract type `/mob/living/basic`, not correct.

File: code/modules/mob/living/basic/farm_animals/bee/_bee.dm [hunk 5]
```diff
@@ -0,0 +1,338 @@
+#define BEE_TRAY_RECENT_VISIT 200 //How long in deciseconds until a tray can be visited by a bee again
```
Comment (san7890) [inline code/modules/mob/living/basic/farm_animals/bee/_bee.dm:1]:
```suggestion
#define BEE_TRAY_RECENT_VISIT 20 SECONDS //! How long until a tray can be visited by a bee again
```

no need for that un-necessary portion of the comment, we can nest defines like this. the `//!` is so it can be dmdocced, but would be nice if you could do a more traditional format for all of these.

File: code/modules/mob/living/basic/farm_animals/bee/_bee.dm [hunk 6]
```diff
@@ -0,0 +1,338 @@
...
+	mob_biotypes = MOB_ORGANIC|MOB_BUG
+	density = FALSE
+	gold_core_spawnable = FRIENDLY_SPAWN
+	can_be_held = TRUE
+	held_w_class = WEIGHT_CLASS_TINY
+	environment_smash  = ENVIRONMENT_SMASH_NONE
+	habitable_atmos = list("min_oxy" = 0, "max_oxy" = 0, "min_plas" = 0, "max_plas" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	basic_mob_flags = DEL_ON_DEATH
+	ai_controller = /datum/ai_controller/basic_controller/bee
+	///he he he, beegent
```
Comment (san7890) [inline code/modules/mob/living/basic/farm_animals/bee/_bee.dm:49]: it's funny but i think you should be more descriptive

File: code/modules/mob/living/basic/farm_animals/bee/_bee.dm [hunk 7]
```diff
@@ -0,0 +1,338 @@
...
+	var/datum/reagent/beegent = null
+	///the house we live in
+	var/obj/structure/beebox/beehome = null
+	///our icon base
+	var/icon_base = "bee"
+	var/isqueen = FALSE
```
Comment (san7890) [inline code/modules/mob/living/basic/farm_animals/bee/_bee.dm:55]: sounds self-explanatory but i'd like if you could DMdoc for statistical purposes. would also prefer if this could be re-named `is_queen` to keep in the `snake_case` standard of this codebase.

File: code/modules/mob/living/basic/farm_animals/bee/_bee.dm [hunk 8]
```diff
@@ -0,0 +1,338 @@
...
+	picker.put_in_hands(holder)
+
+/mob/living/basic/bee/death(gibbed)
+	if(beehome)
+		beehome.bees -= src
+		beehome = null
+	beegent = null
+	return ..()
+
+/mob/living/basic/bee/UnarmedAttack(atom/target, proximity_flag, list/modifiers)
```
Comment (san7890) [inline code/modules/mob/living/basic/farm_animals/bee/_bee.dm:85]: if we're going to override the whole UnarmedAttack proc I think we should use the signal instead. I know there was a bit of a discussion between Melbert and I in another PR but I think if we're going to be killing the entire proc anyhow (and not invoking parent at the start) there shouldn't really be a reason why we shouldn't do this on a signal... any concerns about this though? I don't know if the comsig for attack allows for return types (e.g. some sort of `COMPONENT_CANCEL_SIGNAL` thing)

File: code/modules/mob/living/basic/farm_animals/bee/_bee.dm [hunk 9]
```diff
@@ -0,0 +1,338 @@
...
+		return
+	if(!isnull(hive.queen_bee) && isqueen) //if we are queen and house already have a queen, dont inhabit
+		return
+	if(!hive.habitable(src) || !isnull(beehome)) //if not habitable or we alrdy have a home
+		return
+	beehome = hive
+	beehome.bees += src
+	if(isqueen)
+		beehome.queen_bee = src
+	return
```
Comment (san7890) [inline code/modules/mob/living/basic/farm_animals/bee/_bee.dm:109]:
```suggestion
```

no need for this return.

File: code/modules/mob/living/basic/farm_animals/bee/_bee.dm [hunk 10]
```diff
@@ -0,0 +1,338 @@
...
+
+/mob/living/basic/bee/will_escape_storage()
+	return TRUE
+
+/mob/living/basic/bee/examine(mob/user)
+	. = ..()
+
+	if(!beehome)
```
Comment (san7890) [inline code/modules/mob/living/basic/farm_animals/bee/_bee.dm:117]:
```suggestion
	if(isnull(beehome))
```

faster, more explicit.

File: code/modules/mob/living/basic/farm_animals/bee/_bee.dm [hunk 11]
```diff
@@ -0,0 +1,338 @@
...
+		beehome.bees -= src
+		beehome = null
+	beegent = null
+	return ..()
+
+/mob/living/basic/bee/death(gibbed)
+	if(beehome)
+		beehome.bees -= src
+		beehome = null
+	if((flags_1 & HOLOGRAM_1))
```
Comment (san7890) [inline code/modules/mob/living/basic/farm_animals/bee/_bee.dm:131]:
```suggestion
	if(flags_1 & HOLOGRAM_1)
```

redundant

File: code/modules/mob/living/basic/farm_animals/bee/_bee.dm [hunk 12]
```diff
@@ -0,0 +1,338 @@
...
+		. += span_warning("This bee is homeless!")
+
+/mob/living/basic/bee/Destroy()
+	if(beehome)
+		beehome.bees -= src
+		beehome = null
+	beegent = null
+	return ..()
+
+/mob/living/basic/bee/death(gibbed)
```
Comment (san7890) [inline code/modules/mob/living/basic/farm_animals/bee/_bee.dm:127]: you've defined `death()` twice. the other one is on line 78.

File: code/modules/mob/living/basic/farm_animals/bee/_bee.dm [hunk 13]
```diff
@@ -0,0 +1,338 @@
...
+		beehome = null
+	if((flags_1 & HOLOGRAM_1))
+		return ..()
+	new /obj/item/trash/bee(loc, src)
+	beegent = null
+	return ..()
+
+/mob/living/basic/bee/proc/apply_chemicals(mob/living/basic/attacker, atom/target)
+	SIGNAL_HANDLER
+	if(!beegent || !isliving(target))
```
Comment (san7890) [inline code/modules/mob/living/basic/farm_animals/bee/_bee.dm:139]:
```suggestion
	if(isnull(beegent) || !isliving(target))
```

File: code/modules/mob/living/basic/farm_animals/bee/_bee.dm [hunk 14]
```diff
@@ -0,0 +1,338 @@
...
+		return
+	var/mob/living/victim = target
+	if(victim.reagents)
+		beegent.expose_mob(victim, INJECT)
+		victim.reagents.add_reagent(beegent.type, rand(1,5))
+
+/mob/living/basic/bee/proc/reagent_incompatible(mob/living/basic/bee/ruler)
+	if(!ruler)
+		return FALSE
+	if(ruler.beegent?.type != beegent?.type)
```
Comment (san7890) [inline code/modules/mob/living/basic/farm_animals/bee/_bee.dm:149]: can we do an early return if we have no beegent? just `isnull(beegent)`, given how often we access it.

File: code/modules/mob/living/basic/farm_animals/bee/_bee.dm [hunk 15]
```diff
@@ -0,0 +1,338 @@
...
+		return TRUE
+	if(ruler.beegent && !beegent || !ruler.beegent && beegent)
+		return TRUE
+	return FALSE
+
+/mob/living/basic/bee/proc/generate_bee_visuals()
+	cut_overlays()
+
+	var/col = BEE_DEFAULT_COLOUR
```
Comment (san7890) [inline code/modules/mob/living/basic/farm_animals/bee/_bee.dm:158]:
```suggestion
	var/bee_color = BEE_DEFAULT_COLOUR
```

let's use our words here.

File: code/modules/mob/living/basic/farm_animals/bee/_bee.dm [hunk 16]
```diff
@@ -0,0 +1,338 @@
...
+		hydro.myseed.adjust_yield(1)
+		hydro.yieldmod = 2
+	if(prob(BEE_POLLINATE_POTENCY_CHANCE))
+		hydro.myseed.adjust_potency(1)
+
+	if(beehome)
+		beehome.bee_resources = min(beehome.bee_resources + growth, 100)
+
+/mob/living/basic/bee/proc/assign_reagent(datum/reagent/toxin)
+	if(istype(toxin))
```
Comment (san7890) [inline code/modules/mob/living/basic/farm_animals/bee/_bee.dm:193]: change this to an early return.

File: code/modules/mob/living/basic/farm_animals/bee/_bee.dm [hunk 17]
```diff
@@ -0,0 +1,338 @@
...
+	name = "queen bee"
+	desc = "She's the queen of bees, BZZ BZZ!"
+	icon_base = "queen"
+	isqueen = TRUE
+	ai_controller = /datum/ai_controller/basic_controller/queen_bee
+
+/mob/living/basic/bee/queen/will_escape_storage()
+	return FALSE
+
+/mob/living/basic/bee/toxin
```
Comment (san7890) [inline code/modules/mob/living/basic/farm_animals/bee/_bee.dm:213]: should we add a `desc` or something here? your call.

File: code/modules/mob/living/basic/farm_animals/bee/_bee.dm [hunk 18]
```diff
@@ -0,0 +1,338 @@
...
+	. = ..()
+	queen = new(src)
+
+/obj/item/trash/bee
+	name = "bee"
+	desc = "No wonder the bees are dying out, you monster."
+	icon = 'icons/mob/simple/bees.dmi'
+	icon_state = "bee_item"
+	var/datum/reagent/beegent
+	var/bee_type = /mob/living/basic/bee
```
Comment (san7890) [inline code/modules/mob/living/basic/farm_animals/bee/_bee.dm:293]: dmdoc.

File: code/modules/mob/living/basic/farm_animals/bee/_bee.dm [hunk 19]
```diff
@@ -0,0 +1,338 @@
...
+
+/obj/item/queen_bee/Exited(atom/movable/gone, direction)
+	. = ..()
+	if(gone == queen)
+		queen = null
+		// the bee should not exist without a bee.
+		if(!QDELETED(src))
+			qdel(src)
+
+/obj/item/queen_bee/attackby(obj/item/syringe, mob/user, params)
```
Comment (san7890) [inline code/modules/mob/living/basic/farm_animals/bee/_bee.dm:252]: again, since we override all of `attackby()`, i feel like signals would be optimal?

File: code/modules/mob/living/basic/farm_animals/bee/_bee.dm [hunk 20]
```diff
@@ -0,0 +1,338 @@
...
+	. = ..()
+	AddElement(/datum/element/swabable, CELL_LINE_TABLE_QUEEN_BEE, CELL_VIRUS_TABLE_GENERIC_MOB, 1, 5)
+
+/obj/item/queen_bee/Destroy()
+	QDEL_NULL(queen)
+	return ..()
+
+/obj/item/queen_bee/Exited(atom/movable/gone, direction)
+	. = ..()
+	if(gone == queen)
```
Comment (san7890) [inline code/modules/mob/living/basic/farm_animals/bee/_bee.dm:246]: make this an early return?

File: code/modules/mob/living/basic/farm_animals/bee/_bee.dm [hunk 21]
```diff
@@ -0,0 +1,338 @@
...
+	icon = 'icons/mob/simple/bees.dmi'
+	/// The actual mob that our bee item corresponds to
+	var/mob/living/basic/bee/queen/queen
+
+/obj/item/queen_bee/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/swabable, CELL_LINE_TABLE_QUEEN_BEE, CELL_VIRUS_TABLE_GENERIC_MOB, 1, 5)
+
+/obj/item/queen_bee/Destroy()
+	QDEL_NULL(queen)
```
Comment (san7890) [inline code/modules/mob/living/basic/farm_animals/bee/_bee.dm:241]: damn really?

File: code/modules/mob/living/basic/farm_animals/bee/bee_ai.dm [hunk 22]
```diff
@@ -0,0 +1,175 @@
...
+	return SUBTREE_RETURN_FINISH_PLANNING
+
+
+/datum/ai_planning_subtree/find_and_hunt_target/pollinate
+	target_key = BB_TARGET_HYDRO
+	hunting_behavior = /datum/ai_behavior/hunt_target/unarmed_attack_target/bees
+	finding_behavior = /datum/ai_behavior/find_hunt_target/pollinate
+	hunt_targets = list(/obj/machinery/hydroponics)
+	hunt_range = 7
+	//hunt_chance = 10
```
Comment (san7890) [inline code/modules/mob/living/basic/farm_animals/bee/bee_ai.dm:66]: no commented out code like this, either add it in fully or delete the line.

File: code/modules/mob/living/basic/farm_animals/bee/bee_ai.dm [hunk 23]
```diff
@@ -0,0 +1,175 @@
...
+
+
+/datum/ai_behavior/hunt_target/unarmed_attack_target/bees
+	always_reset_target = TRUE
+
+/datum/ai_behavior/find_hunt_target/pollinate
+
+/datum/ai_behavior/find_hunt_target/pollinate/valid_dinner(mob/living/source, obj/machinery/hydroponics/dinner, radius)
+	if(!dinner.myseed)
```
Comment (san7890) [inline code/modules/mob/living/basic/farm_animals/bee/bee_ai.dm:75]: `isnull()`?

File: code/modules/mob/living/basic/farm_animals/bee/bee_ai.dm [hunk 24]
```diff
@@ -0,0 +1,175 @@
...
+	if(QDELETED(target))
+		return FALSE
+	set_movement_target(controller, target)
+
+/datum/ai_behavior/enter_exit_hive/perform(seconds_per_tick, datum/ai_controller/controller, target_key)
+	. = ..()
+	var/obj/structure/beebox/current_home = controller.blackboard[target_key]
+	var/mob/living/bee_pawn = controller.pawn
+
+	bee_pawn.UnarmedAttack(current_home)
```
Comment (san7890) [inline code/modules/mob/living/basic/farm_animals/bee/bee_ai.dm:97]: Unarmed attack? I think we should split that behavior out of UnarmedAttack out into a new board and then store it as a `/datum/callback` in the blackboard or something that we can execute since we definitely know that this is going to execute the enter hive behavior. Going to have really odd consequences if this behavior is drag and dropped anywhere else. Would allow it to be infinitely more extensible to more mobs that might need to pop in and out of hives without being reliant on unarmed attack.

File: code/modules/mob/living/basic/farm_animals/bee/bee_ai.dm [hunk 25]
```diff
@@ -0,0 +1,175 @@
...
+/datum/ai_behavior/inhabit_hive/perform(seconds_per_tick, datum/ai_controller/controller, target_key)
+	. = ..()
+	var/obj/structure/beebox/potential_home = controller.blackboard[target_key]
+	var/mob/living/bee_pawn = controller.pawn
+
+	if(!potential_home.habitable(bee_pawn)) //the house become full before we get to it
+		finish_action(controller, FALSE, target_key)
+		return
+
+	bee_pawn.UnarmedAttack(potential_home) //interact with the house to inhabit
```
Comment (san7890) [inline code/modules/mob/living/basic/farm_animals/bee/bee_ai.dm:119]: ditto on splitting it out of UnarmedAttack and into a callback proc.

File: code/modules/mob/living/basic/farm_animals/bee/bee_ai.dm [hunk 26]
```diff
@@ -0,0 +1,175 @@
...
+
+	if(valid_hives.len)
+		return pick(valid_hives)
+
+/datum/targetting_datum/basic/bee
+
+/datum/targetting_datum/basic/bee/can_attack(mob/living/owner, atom/target)
+	if(!isliving(target))
+		return FALSE
+	. = ..()
```
Comment (san7890) [inline code/modules/mob/living/basic/farm_animals/bee/bee_ai.dm:150]: isn't there a `pre_attack` check proc for this? why do we invoke parent in the middle?

**Review by san7890 — CHANGES_REQUESTED** [review]
Mostly nits, although a few serious code issues were noted.

**Comment by san7890** [comment]
lol wait why didn't github show me this was drafted? sorry, still should address a good chunk of this stuff though.

File: code/modules/mob/living/basic/farm_animals/bee/bee_ai.dm (hunk 24 above)
Comment (Ben10Omintrix) [reply code/modules/mob/living/basic/farm_animals/bee/bee_ai.dm:97]: i putted him in the unarmedatack so if the bee is sentient he can interacted with the objects the ai can interacted with

File: code/modules/mob/living/basic/farm_animals/bee/_bee.dm (hunk 14 above)
Comment (Ben10Omintrix) [reply code/modules/mob/living/basic/farm_animals/bee/_bee.dm:149]: the bee can entered the hive even if he dont have a beegent

File: code/modules/mob/living/basic/farm_animals/bee/bee_ai.dm (hunk 26 above)
Comment (Ben10Omintrix) [reply code/modules/mob/living/basic/farm_animals/bee/bee_ai.dm:150]: i cant find this proc

File: code/modules/hydroponics/beekeeping/beebox.dm [hunk 27]
```diff
@@ -47,9 +44,13 @@
 
 /obj/structure/beebox/Destroy()
 	STOP_PROCESSING(SSobj, src)
-	bees.Cut()
-	honeycombs.Cut()
 	queen_bee = null
+	for(var/mob/living/basic/bee/worker in bees)
+		bees -= worker
+		worker.beehome = null
+		worker.forceMove(get_turf(src))
```
Comment (MrMelbert) [inline code/modules/hydroponics/beekeeping/beebox.dm:51]: `Destroy` should not have side effects, any remaining workers should be qdeleted (which is what is currently being done, any bees in the contents of the box are qdel'd)

You can however also implement `/deconstruct` and put this code here, so if someone bashes the beebox to bits, the bees will flow out instead.

File: code/modules/hydroponics/beekeeping/beebox.dm [hunk 28]
```diff
@@ -198,16 +199,14 @@
 	. = ..()
 	if(!user.bee_friendly())
 		//Time to get stung!
-		var/bees = FALSE
-		for(var/b in bees) //everyone who's ever lived here now instantly hates you, suck it assistant!
-			var/mob/living/simple_animal/hostile/bee/B = b
-			if(B.isqueen)
+		var/bees_attack = FALSE
+		for(var/mob/living/basic/bee/worker in bees) //everyone who's ever lived here now instantly hates you, suck it assistant!
```
Comment (MrMelbert) [inline code/modules/hydroponics/beekeeping/beebox.dm:203]: `as anything`, if the bees list is only bees

File: code/modules/hydroponics/beekeeping/beebox.dm [hunk 29]
```diff
@@ -255,9 +254,9 @@
 
 /obj/structure/beebox/deconstruct(disassembled = TRUE)
 	new /obj/item/stack/sheet/mineral/wood (loc, 20)
-	for(var/mob/living/simple_animal/hostile/bee/B in bees)
-		if(B.loc == src)
-			B.forceMove(drop_location())
+	for(var/mob/living/basic/bee/worker in bees)
```
Comment (MrMelbert) [inline code/modules/hydroponics/beekeeping/beebox.dm:257]: `as anything`

File: code/modules/mob/living/basic/farm_animals/bee/_bee.dm [hunk 30]
```diff
@@ -0,0 +1,338 @@
...
+	beegent = null
+	return ..()
+
+/mob/living/basic/bee/death(gibbed)
+	if(beehome)
+		beehome.bees -= src
+		beehome = null
+	if(flags_1 & HOLOGRAM_1)
+		return ..()
+	new /obj/item/trash/bee(loc, src)
```
Comment (MrMelbert) [inline code/modules/mob/living/basic/farm_animals/bee/_bee.dm:126]: Small thing but it shouldn't drop bee trash if `gibbed` is `TRUE`

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/basic/farm_animals/bee/_bee.dm (hunk 8 above)
Comment (MrMelbert) [reply code/modules/mob/living/basic/farm_animals/bee/_bee.dm:79]: I don't think there's any immediately wrong with overriding `UnarmedAttack` since it's the highest level part of the click chain you can get on the mob without going directly into Click

(However you should check proximity flag, it's only set by telekinesis but you never know)

That being said apparently basic mobs do have `/melee_attack` which is called from `UnarmedAttack` which might be more appropriate to override-while-conditionally-calling-parent

File: code/modules/mob/living/basic/farm_animals/bee/bee_ai.dm (hunk 26 above)
Comment (san7890) [reply code/modules/mob/living/basic/farm_animals/bee/bee_ai.dm:150]: i could've sworn there was something we check in the parent of `can_attack`. sigh

**Comment by san7890** [comment]
i think you forgot to push the commits that addressed melbert's reviews, i unmarked them

File: code/modules/mob/living/basic/farm_animals/bee/_bee.dm (hunk 14 above)
Comment (san7890) [reply code/modules/mob/living/basic/farm_animals/bee/_bee.dm]: This is `reagent_incompatible()` though? what's that got to do with this?

File: code/modules/mob/living/basic/farm_animals/bee/bee_ai.dm (hunk 24 above)
Comment (san7890) [reply code/modules/mob/living/basic/farm_animals/bee/bee_ai.dm:97]: Forgot to elaborate, this would be done in conjunction with the other suggestion that the earlier behavior should be split out into a signal (or at least invoked via `melee_attack()`) so that both the player bee and the AI bee can use it by either of those interactions.

File: code/modules/mob/living/basic/farm_animals/bee/_bee.dm (hunk 8 above)
Comment (san7890) [reply code/modules/mob/living/basic/farm_animals/bee/_bee.dm:79]: i would be amenable to either `melee_attack()` or a signal handler for this then, i just don't like completely glonking the click chain which is going to always be subject to a bunch of bullshit in the past, present, and future.

File: code/modules/mob/living/basic/farm_animals/bee/_bee.dm (hunk 14 above)
Comment (san7890) [reply code/modules/mob/living/basic/farm_animals/bee/_bee.dm]: oh you changed it in the code nvm then

File: code/modules/mob/living/basic/farm_animals/bee/bee_ai.dm (hunk 26 above)
Comment (san7890) [reply code/modules/mob/living/basic/farm_animals/bee/bee_ai.dm:150]: or is `can_attack()` the pre-check code itself? ugh

**Comment by Ben10Omintrix** [comment]
> i think you forgot to push the commits that addressed melbert's reviews, i unresolved them for the time being. I also addressed your comments and elaborated on some stuff I wrote.

i am finished the changes u want and i tested him and its working. u can check the code again?

File: code/modules/mob/living/basic/farm_animals/bee/_bee.dm [hunk 31]
```diff
@@ -0,0 +1,339 @@
...
+		pixel_x = dead_bee.pixel_x
+		pixel_y = dead_bee.pixel_y
+		bee_type = dead_bee.type
+		if(dead_bee.beegent)
+			beegent = dead_bee.beegent
+			reagents.add_reagent(beegent.type, 5)
+		update_appearance()
+	AddComponent(/datum/component/edible, list(/datum/reagent/consumable/nutriment/vitamin = 5), null, BEE_FOODGROUPS, 10, 0, list("bee"), null, 10)
+	AddElement(/datum/element/swabable, CELL_LINE_TABLE_QUEEN_BEE, CELL_VIRUS_TABLE_GENERIC_MOB, 1, 5)
+	RegisterSignal(src, COMSIG_ATOM_ON_LAZARUS_INJECTOR, PROC_REF(use_lazarus))
```
Comment (san7890) [inline code/modules/mob/living/basic/farm_animals/bee/_bee.dm:308]: reorganize this entire proc to be an early return (so just do the whole addcomponent/element/signal stuff first, then early return on an `if(isnull(dead_bee))`

**Review by san7890 — CHANGES_REQUESTED** [review]
my last thoughts on the matter

File: code/modules/mob/living/basic/farm_animals/bee/bee_ai.dm [hunk 32]
```diff
@@ -0,0 +1,181 @@
+
+
+/datum/ai_controller/basic_controller/bee
```
Comment (san7890) [inline code/modules/mob/living/basic/farm_animals/bee/bee_ai.dm:3]: i also think this whole implementation would look a lot cleaner if you split this file up into multiple files i.e. `bee_subtrees.dm`, `bee_behavior.dm`, etc. seeing as we already have a full folder for bee stuff already.

File: code/modules/mob/living/basic/farm_animals/bee/_bee.dm [hunk 33]
```diff
@@ -130,15 +129,6 @@
 	if(is_queen)
 		beehome.queen_bee = src
 
-/mob/living/basic/bee/proc/apply_chemicals(mob/living/basic/attacker, atom/target)
```
Comment (san7890) [inline code/modules/mob/living/basic/farm_animals/bee/_bee.dm]: why was this removed, is it done elsewhere?

Comment (Ben10Omintrix) [reply code/modules/mob/living/basic/farm_animals/bee/_bee.dm]: yes. he get a venemous element in assign_reagent() so this is dont need now
